rest-service/manager_rest/resource_manager.py

```diff
@@ -2317,13 +2370,37 @@ def _create_deployment_initial_dependencies(self,
                             fail_silently=True) if target_deployment else None
 
             now = utils.get_formatted_timestamp()
+            if external_client:
+                external_target = {
+                    'deployment': (target_deployment_config.get('id') if
+                                   target_deployment_config else None),
+                    'client_config': client_config
+                }
+
             self.sm.put(models.InterDeploymentDependencies(
                 id=str(uuid.uuid4()),
                 dependency_creator=func_id,
                 source_deployment=source_deployment,
-                target_deployment=target_deployment_instance,
+                target_deployment=(target_deployment_instance if
```

Comment:
Suggestion: `None if external_client else target_deployment_instance`, putting a `not` there just makes it more complicated. This is irrelevant if you go with the previous comment.